bip-0054.md

```diff
@@ -0,0 +1,231 @@
+this effect a limit on the number of potentially executed signature operations pinpoints exactly the
+harmful behaviour, leaving maximum flexibility in how Script functionalities may have been used.
+Such a limit reduces the worst case block validation time by a factor of 40 and drastically
+increases the preparation cost of an attack to make it uneconomical for a miner[^6]. The maximum of
+2500 was chosen as the tightest value that did not make any non-pathological standard transaction
+invalid[^7].
+
+In the presence of 64-byte transactions a block header's Merkle root may be valid for different sets
+of transactions. This is because in the Merkle tree construction a 64-byte transaction may be
+interpreted as the catenation of two 32-byte hashes, or the catenation of two 32-byte hashes may be
+interpreted as a transaction. The former allows to fake a block inclusion proof and the latter makes
+it such that for a valid block the Merkle root in the block header is not a unique identifier for
+the corresponding list of valid transactions[^8]. 64-byte transactions can only contain a
+scriptPubKey that lets anyone spend the funds, or one that burns them. 64-byte transactions have
+also been non-standard since 2019. It was suggested that the known vulnerabilities could instead be
+mitigated by committing to the Merkle tree depth in the header's version field[^9]. The authors
+believe it is preferable to address the root cause by invalidating 64-byte transactions. This
+approach also fixes the vulnerability without developers of SPV verifiers having to implement the
+mitigation or to know it is necessary in the first place.
+
+Several blocks prior to [bip-0034][BIP34] activation contain a coinbase transaction whose scriptSig
+contains a valid [bip-0034][BIP34] commitment to a future block height. This offers an opportunity
+to duplicate these coinbase transactions in the future[^10] and for this reason [bip-0030][BIP30]
+validation will need to be re-activated from block 1,983,702. A simple way to prevent this is to
+mandate that future coinbase transactions vary from coinbase transactions before [bip-0034][BIP34]
+activation. There are multiple ways of achieving this, but setting and enforcing the timelock for
+the coinbase transaction makes it so all coinbase transactions past Consensus Cleanup activation
+could not have been valid before this height and therefore cannot be a duplicate[^11].
+
+## Backward compatibility
+
+This proposal only tightens the block validation rules: there is no block that is valid under the
+rules proposed in this BIP but not under the existing Bitcoin consensus rules. As a consequence
+these changes are backward-compatible with unupgraded node software. That said, the authors strongly
+encourage node operators to upgrade in order to fully validate all consensus rules.
+
+## Miner forward compatibility
+
+Bitcoin Core version [0.16.1][Core 0.16.1] and later will not relay nor create block templates which
```

Comment:
Use either "neither...nor" or "not...or", and "that" instead of "which"

```suggestion
Bitcoin Core version [0.16.1][Core 0.16.1] and later will neither relay nor create block templates that
```

Reply:
Done.